**Nomes e RA**

Felipe Ferreira de Medeiros
3019101810

Cintia Regina da Conceicao Lopes
3022100478

Rafael dos Santos Silva
3022103882

Emily Onofre Rodrigues
3024106554

Henrique dos Santos Moreira
3022107720

Leandro Cesarini Alves Batista
3022203187

Gustavo Henrique B Dias
3022201083

Lucas Lunardi Barros
3022102523

Lucas Correa de Jesus
3024106885

Wagner Rodrigues
3022201803

In [1]:
# Conexão com o drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# Carrega a biblioteca pandas e atribui um alias chamado "pd"
import pandas as pd

# Visualização de dados
import matplotlib.pyplot as plt
import seaborn as sns

# Deixar colunas e linhas aparentes
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [3]:
#Faz a leitura do arquivo igm_modificado.csv - alterar o caminho de onde está dua tabela
df = pd.read_csv('/content/drive/MyDrive/salários - salários.csv')
df.shape

(3771, 13)

In [4]:
# Verificar as colunas do DataFrame
colunas = df.columns
print(colunas)

Index(['Unnamed: 0', 'work_year', 'experience_level', 'employment_type',
       'job_title', 'salary', 'salary_currency', 'salary_in_usd',
       'employee_residence', 'remote_ratio', 'company_location',
       'company_size', 'country_employee'],
      dtype='object')


In [5]:
# Exercício 1

In [6]:
# Definindo o valor de referência para comparação
salario_referencia = 15000

# Contadores para acompanhar quantos salários estão acima, abaixo ou igual ao valor de referência
acima_referencia = 0
abaixo_referencia = 0
igual_referencia = 0

# Verificando o salário em relação ao valor de referência
for salario in df['salary']:
    if salario > salario_referencia:
        acima_referencia += 1
    elif salario < salario_referencia:
        abaixo_referencia += 1
    else:
        igual_referencia += 1

# Imprimindo os resultados
print(f"Salários acima de {salario_referencia}: {acima_referencia}")
print(f"Salários abaixo de {salario_referencia}: {abaixo_referencia}")
print(f"Salários iguais a {salario_referencia}: {igual_referencia}")

Salários acima de 15000: 3742
Salários abaixo de 15000: 25
Salários iguais a 15000: 4


Contexto da Análise:

Analisamos os salários dos profissionais de dados registrados na tabela. Nosso objetivo foi entender como a distribuição dos salários se relaciona com um valor de referência de 15000. Assim, podemos identificar quantos profissionais ganham acima, abaixo ou igual a esse valor estabelecido, o que ajuda a fornecer ideias bases sobre a faixa salarial dos profissionais de dados.

Análise dos Resultados:

A análise revela que a maioria dos profissionais de dados ganha acima de 15000, com 3742 salários nesta faixa. Isso nos mostra que há uma alta remuneração na área de informática, muito provavelmente, devido à demanda por habilidades em ciência e análise de dados. No entanto, cerca de 25 profissionais ganham abaixo desse valor, o que sugere variações salariais. A presença de 4 salários iguais a 15000 unidades pode indicar faixas salariais padrão em algumas empresas.

In [7]:
# Exercício 2

In [8]:
# Contagem do número de ocorrências para cada nível de experiência
contagem_experiencia = df['experience_level'].value_counts()

# Imprimindo as informações da coluna 'experience_level'
print(contagem_experiencia)

experience_level
SE    2520
MI     811
EN     324
EX     116
Name: count, dtype: int64


In [9]:
# Contagens fornecidas para cada nível de experiência
contagem_experiencia = {
    'SE': 2520,
    'MI': 811,
    'EN': 324,
    'EX': 116
}

# Definindo as categorias com base no nível de experiência
senior = contagem_experiencia.get('SE', 0)
pleno = contagem_experiencia.get('MI', 0)
junior = contagem_experiencia.get('EN', 0)
especialista = contagem_experiencia.get('EX', 0)

# Imprimindo as categorias e seus respectivos números de ocorrências
print("Número de funcionários por nível de experiência:")
print("SE (Sênior):", senior)
print("MI (Pleno):", pleno)
print("EN (Junior):", junior)
print("EX (Especialista):", especialista)

# Desvio condicional encadeado para análise dos resultados
if senior > pleno and senior > junior and  senior > especialista:
    print("A maioria dos funcionários está no nível Sênior.")
elif pleno > senior and pleno > junior and pleno > especialista:
    print("A maioria dos funcionários está no nível Pleno.")
elif junior > senior and junior > pleno and junior > especialista:
    print("A maioria dos funcionários está no nível Junior.")
elif especialista > senior and especialista > pleno and especialista > junior:
    print("A maioria dos funcionários está no nível Especialista.")
else:
    print("Não há um nível de experiência dominante entre os funcionários.")

Número de funcionários por nível de experiência:
SE (Sênior): 2520
MI (Pleno): 811
EN (Junior): 324
EX (Especialista): 116
A maioria dos funcionários está no nível Sênior.


Contexto da Análise:

Esta analise de dados foi feita para medir os níveis de experiencia dos cientistas de dados, e como é distribuída para as equipes junior, pleno, sênior e especialistas.

Análise dos Resultados:

Os resultados mostram que a maioria dos cientistas de dados estão nos níveis Sênior (SE) e Pleno (MI). Isso sugere que há uma quantidade significativa de profissionais de dados com experiência intermediária/avançada nas equipes. No entanto, também há um número considerável de cientistas de dados no nível Júnior (EN), o que indica a presença de profissionais em estágios iniciais de suas carreiras na área. O número de cientistas de dados no nível Especialista (EX) é menor em relação aos outros

In [10]:
# Exercício 3

In [11]:
contagem_por_pais = {}

for pais in df['company_location']:
    if pais in contagem_por_pais:
        contagem_por_pais[pais] += 1
    else:
        contagem_por_pais[pais] = 1

# Mostrar os resultados
print("Distribuição de Empregadores por País:")
for pais, contagem in contagem_por_pais.items():
    print(f"{pais}: {contagem} profissionais de dados")

Distribuição de Empregadores por País:
ES: 77 profissionais de dados
US: 3046 profissionais de dados
CA: 87 profissionais de dados
DE: 56 profissionais de dados
GB: 173 profissionais de dados
NG: 5 profissionais de dados
IN: 58 profissionais de dados
HK: 1 profissionais de dados
NL: 13 profissionais de dados
CH: 5 profissionais de dados
CF: 2 profissionais de dados
FR: 34 profissionais de dados
FI: 3 profissionais de dados
UA: 4 profissionais de dados
IE: 7 profissionais de dados
IL: 2 profissionais de dados
GH: 2 profissionais de dados
CO: 4 profissionais de dados
SG: 6 profissionais de dados
AU: 14 profissionais de dados
SE: 2 profissionais de dados
SI: 4 profissionais de dados
MX: 10 profissionais de dados
BR: 15 profissionais de dados
PT: 14 profissionais de dados
RU: 5 profissionais de dados
TH: 6 profissionais de dados
HR: 3 profissionais de dados
VN: 2 profissionais de dados
EE: 3 profissionais de dados
AM: 1 profissionais de dados
BA: 1 profissionais de dados
KE: 2 profissionai

Contexto da Análise:

Verificamos a distribuição dos empregadores de profissionais de dados em diferentes países. A coluna 'company_location' contém informações sobre o país onde está localizada a sede ou filial do empregador de cada profissional de dados. O objetivo é entender como a demanda por profissionais de dados está distribuída globalmente e identificar os países com uma presença significativa nesse campo.

Análise dos Resultados:

Os Estados Unidos (US) lideram em número de empregadores de profissionais de dados, com 3.046 profissionais de dados registrados.
Em segundo lugar, o Reino Unido (GB) com 173 profissionais de dados, seguido pelo Canadá (CA) com 87 profissionais de dados.
Além dos EUA e do Reino Unido, outros países com uma presença considerável de empregadores de profissionais de dados incluem Espanha (ES) com 77 profissionais de dados, Índia (IN) com 58 e Alemanha (DE) com 56.
Alguns países têm um número muito baixo de empregadores de profissionais de dados, como Hong Kong (HK) com apenas 1 profissional de dados, enquanto outros tem apenas algumas empresas representadas.

Os resultados mostram de forma global a demanda por profissionais de dados, com os Estados Unidos liderando, mas com uma presença significativa em vários outros países ao redor do mundo. Essa análise e informações pode ser útil para profissionais que buscam oportunidades de emprego em diferentes regiões e para empresas que desejam entender melhor a concorrência e a disponibilidade de talentos em diferentes localidade.

In [12]:
# Exercício 4

In [13]:
contagem_emprego = {}

indice = 0

total_linhas = len(df)

while indice < total_linhas:

    tipo_emprego = df['employment_type'][indice]

    if tipo_emprego in contagem_emprego:

        contagem_emprego[tipo_emprego] += 1
    else:

        contagem_emprego[tipo_emprego] = 1

    indice += 1

# Imprimindo os resultados
print("Distribuição dos tipos de emprego entre os profissionais de dados:")
for tipo, contagem in contagem_emprego.items():
    print(f"{tipo}: {contagem} profissionais de dados")

Distribuição dos tipos de emprego entre os profissionais de dados:
FT: 3731 profissionais de dados
CT: 11 profissionais de dados
FL: 11 profissionais de dados
PT: 18 profissionais de dados


Contexto da Análise:

Analisamos a distribuição dos tipos de emprego entre os profissionais de dados. A coluna 'employment_type' nos fornece informações sobre o tipo de emprego de cada profissional durante o respectivo ano. Os tipos de emprego incluem trabalho em período integral, contratual, meio período e freelancer, representados por códigos 'FT', 'CT', 'PT' e 'FL'. Nosso objetivo é compreender como esses diferentes tipos de emprego estão distribuídos entre os profissionais de dados.

Análise dos Resultados:

Através da análise exploratória foi revelado que a maioria dos profissionais de dados trabalha em período integral, depois meio período, seguido por contratual,  e freelancers.

São informações que podem ser valiosas para empresas e recrutadores que buscam entender as preferências de emprego dos profissionais de dados e adaptar suas estratégias de contratação de talentos. Além disso, os profissionais de dados podem usar esses insights para avaliar a demanda por diferentes tipos de emprego.

_______________________________________________________________________________


In [14]:
#Exercício 5

In [15]:
import pandas as pd

# Função para analisar a coluna remote_ratio
def analyze_remote_ratio(dataframe):
    # Calcula a média usando uma função lambda
    mean_remote_ratio = dataframe['remote_ratio'].apply(lambda x: x).mean()

    # Calcula a mediana usando uma função lambda
    median_remote_ratio = dataframe['remote_ratio'].apply(lambda x: x).median()

    # Calcula o desvio padrão usando uma função lambda
    std_remote_ratio = dataframe['remote_ratio'].apply(lambda x: x).std()

    # Retorna os resultados
    return mean_remote_ratio, median_remote_ratio, std_remote_ratio

# Supondo que 'df' seja o seu DataFrame
mean_remote_ratio, median_remote_ratio, std_remote_ratio = analyze_remote_ratio(df)

print("Estatísticas da coluna remote_ratio:")
print(f"Média: {mean_remote_ratio}")
print(f"Mediana: {median_remote_ratio}")
print(f"Desvio Padrão: {std_remote_ratio}")

Estatísticas da coluna remote_ratio:
Média: 46.35375232033943
Mediana: 0.0
Desvio Padrão: 48.55297113950274


Contexto de Análise:

Fizemos a analise da coluna 'remote_ratio', pois da pra ter uma boa ideia sobre a proporção de trabalho remoto em diferentes posições que estão nos dados. Com a pandemia, muitas empresas tomaram como medida o home office em varios cargos diferentes, e fazer essa analise podemos ter insights sobre essa proporção.

Análise dos Resultados:

Verificamos que a média de 46.35%, nos informa que quase a metade do trabalho, pode ser realizado remotamente.
A mediana em 0.0%, nos mostra que metade das posições de trabalho, não possui trabalho remoto. Com isso podemos imaginar que essa falta de trabalho remoto em muitas posições, pode ocorrer devido a politica da empresa, requisito do cargo etc.
O desvio padrão de 48.55, mostra uma grande dispersão dos valores em relação a média, com isso podemos verificar uma variabilidade significativa nos níveis de trabalho remoto.


In [16]:
# Exercício 6

In [17]:
import pandas as pd

def analyze_salary_currency(dataframe):
    # Calcula a contagem de cada moeda
    currency_counts = dataframe['salary_currency'].value_counts()

    # Calcula a proporção de cada moeda em relação ao total
    currency_proportion = currency_counts / len(dataframe)

    # Retorna os resultados
    return currency_counts, currency_proportion

# Supondo que 'df' seja o seu DataFrame
currency_counts, currency_proportion = analyze_salary_currency(df)

print("Contagem de cada moeda de salário:")
print(currency_counts)
print("\nProporção de cada moeda de salário em relação ao total:")
print(currency_proportion)

Contagem de cada moeda de salário:
salary_currency
USD    3237
EUR     237
GBP     161
INR      60
CAD      25
AUD       9
SGD       6
BRL       6
PLN       5
CHF       4
THB       4
HUF       3
DKK       3
JPY       3
TRY       3
ILS       1
HKD       1
CZK       1
MXN       1
CLP       1
Name: count, dtype: int64

Proporção de cada moeda de salário em relação ao total:
salary_currency
USD    0.858393
EUR    0.062848
GBP    0.042694
INR    0.015911
CAD    0.006630
AUD    0.002387
SGD    0.001591
BRL    0.001591
PLN    0.001326
CHF    0.001061
THB    0.001061
HUF    0.000796
DKK    0.000796
JPY    0.000796
TRY    0.000796
ILS    0.000265
HKD    0.000265
CZK    0.000265
MXN    0.000265
CLP    0.000265
Name: count, dtype: float64


Contexto da Análise:

Analisamos a coluna 'salary_currency' para entender a distribuição das moedas em que os salários estão nesse conjunto de dados. COm isso, podemos verificar e obter insights sobre a localização e as preferências de moeda de pagamento em diferentes regiões.

Análise dos Resultados:

Vericicamos que a maioria dos salários são em doláres americanos (USD), com um total de 3237, isso nos mostra que existe uma presença forte de empregos e empresa no EUA.
Em seguida, a moeda mais comum é o euro (EUR) com 237, seguido pela libra (GBP) com 161.
Com isso notamos que existe uma força muito grande de mercado no EUA, e a presença de outras moedas nos mostra uma diversidade geográfica, mas a influência do dolar americano ainda é a mais dominante.

In [19]:
# Exercício 7

In [20]:
import pandas as pd

# Carregar os dados do arquivo CSV
dados = pd.read_csv('/content/drive/MyDrive/salários - salários.csv')

# Calculando Remote Ratio
remote_ratio = (dados['remote_ratio'].sum() / len(dados)) * 100

# Calculando a Média Salarial em USD
media_salarial_usd = dados['salary_in_usd'].mean()

print("Remote Ratio:", remote_ratio)
print("Média Salarial (em USD):", media_salarial_usd)

Remote Ratio: 4635.375232033944
Média Salarial (em USD): 137262.1646778043


Contexto da Análise:

Analisamos que é fornecido informações sobre diferentes variáveis relacionadas aos funcionários, incluindo dados sobre trabalho remoto, localização da empresa, tamanho da empresa, etc.
Com isso, nosso objetivo é entender melhor a dinâmica da força de trabalho e o nível de remuneração na empresa.
Análise dos Resultados:

Remote Ratio:

O Remote Ratio calcula a proporção de funcionários que trabalham remotamente em relação ao total de funcionários.
Um valor de Remote Ratio mais alto pode indicar uma tendência de trabalho remoto na empresa.
Por exemplo, se o Remote Ratio for 30%, isso significa que cerca de 30% dos funcionários trabalham remotamente.
Isso pode indicar uma política de trabalho flexível na empresa ou uma resposta à pandemia COVID-19.
Média Salarial em USD:

A Média Salarial em USD calcula a média dos salários dos funcionários em dólares americanos.
Isso pode nos fornecer insights sobre o nível de remuneração na empresa em comparação com outras empresas.
Por exemplo, se a média salarial for $50,000 USD, isso significa que, em média, os funcionários ganham $50,000 USD por ano.
Essa análise pode ser usada para avaliar a competitividade salarial da empresa e sua capacidade de atrair e segurar talentos.

Análise dos Resultados:

A análise do Remote Ratio e Média Salarial fornece insights valiosos sobre a dinâmica da força de trabalho e o nível de remuneração na empresa.
Com esses resultados, a empresa pode considerar ajustes em suas políticas de trabalho remoto e estratégias de remuneração para melhorar a satisfação dos funcionários e a competitividade no mercado.